### PR TITLE
Frio upgrade fontawesome

### PR DESCRIPTION
**Pull request by vinzv**
As a followup to #7542 this is the full migration from fontawesome v4.x to v5.x (v5.10.1 as of 2019-08-21). Steps I mare are:

1. I grepped through all Frio templates, CSS and JS files to catch every icon used. 
2. Afterwards I replace all of them with [this little python script](https://github.com/vinzv/fontawesome4to5).
3. Grepped again to get any missing icons and special cases, fixing them manually.
4. Remove old fontawesome 4.x files, added new 5.x ones.
5. Fixed manually used icons in CSS pseudo elements.
6. Adapted a few icons to look more like their v4.x predecessors.
7. Visual checks by clicking through all pages on my test node.

### CODE REVIEW

**Comment by redmatrix** [comment]
Slightly off-topic but might be relevant. If I recall correctly there were license changes with the latest fontawesome which one may need explore. I don't have the details in front of me. Many other projects have moved to fork-awesome which is a continuation of 4.x but with some nice goodies like icon font versions of most of the major fediverse logos (including Friendica).

File: view/theme/frio/css/style.css [hunk 1]
```diff
@@ -1517,11 +1517,13 @@ textarea.comment-edit-text:focus + .comment-edit-form .preview {
 }
 
 .fa.lock:before {
-    font-family: FontAwesome;
+    font-family: "Font Awesome 5 Free";
+	font-weight: 900;
```
Comment (MrPetovan) [inline view/theme/frio/css/style.css]: Code standards: Please convert tab to 4 spaces.

File: view/theme/frio/css/style.css [hunk 2]
```diff
@@ -1517,11 +1517,13 @@ textarea.comment-edit-text:focus + .comment-edit-form .preview {
...
     content: "\f023";
 }
 .fa.unlock:before {
-    font-family: FontAwesome;
+    font-family: "Font Awesome 5 Free";
+	font-weight: 900;
```
Comment (MrPetovan) [inline view/theme/frio/css/style.css]: Code standards: Please convert tab to 4 spaces.

File: view/theme/frio/css/style.css [hunk 3]
```diff
@@ -2906,12 +2908,14 @@ ul li:hover .contact-wrapper .contact-action-link:hover {
     padding: 1px 10px;
 }
 .panel .section-subtitle-wrapper a.accordion-toggle:before {
-    font-family: FontAwesome;
+    font-family: "Font Awesome 5 Free";
+	font-weight: 900;
```
Comment (MrPetovan) [inline view/theme/frio/css/style.css]: Code standards: Please convert tab to 4 spaces.

File: view/theme/frio/css/style.css [hunk 4]
```diff
@@ -2906,12 +2908,14 @@ ul li:hover .contact-wrapper .contact-action-link:hover {
...
     content: "\f0d7";
     padding-right: 5px;
 }
 .panel .section-subtitle-wrapper a.accordion-toggle.collapsed:before {
-    font-family: FontAwesome;
+    font-family: "Font Awesome 5 Free";
+	font-weight: 900;
```
Comment (MrPetovan) [inline view/theme/frio/css/style.css]: Code standards: Please convert tab to 4 spaces.

File: view/theme/frio/css/style.css [hunk 5]
```diff
@@ -2956,11 +2960,13 @@ ul li:hover .contact-wrapper .contact-action-link:hover {
     padding-right: 10px;
 }
 .settings-block.fakelink > h3:before {
-    font-family: FontAwesome;
+    font-family: "Font Awesome 5 Free";
+	font-weight: 900;
```
Comment (MrPetovan) [inline view/theme/frio/css/style.css]: Code standards: Please convert tab to 4 spaces.

File: view/theme/frio/css/style.css [hunk 6]
```diff
@@ -2956,11 +2960,13 @@ ul li:hover .contact-wrapper .contact-action-link:hover {
...
     content: "\f0da"; /* Right Plain Pointer */
 }
 .settings-block > .fakelink > h3:before {
-    font-family: FontAwesome;
+    font-family: "Font Awesome 5 Free";
+	font-weight: 900;
```
Comment (MrPetovan) [inline view/theme/frio/css/style.css]: Code standards: Please convert tab to 4 spaces.

File: view/theme/frio/css/style.css [hunk 7]
```diff
@@ -3153,7 +3159,8 @@ section.help-content-wrapper li {
     transition: border 0.15s ease-in-out, color 0.15s ease-in-out;
 }
 #adminpage li .icon.on:after {
-    font-family: "FontAwesome";
+    font-family: "Font Awesome 5 Free";
+	font-weight: 900;
```
Comment (MrPetovan) [inline view/theme/frio/css/style.css]: Code standards: Please convert tab to 4 spaces.

**Comment by vinzv** [comment]
Yes, I remember there was some trouble and concerns. The main reasons were afaik that FontAwesome a) is not fully free anymore and b) weren't accepting PRs at some time.
Both points would not keep me back from using it, but I'm open to getting convinced differently.

**Comment by annando** [comment]
Using fork-awesome (when it does provide everything that we need) and the composer (to avoid having third party stuff in our repo) sounds both good in my ears.

P.S.: Thanks for your work in the last days!

**Comment by vinzv** [comment]
Having slept over the idea of migration I thought it really might be better to stick with Fork Awesome. I'll open a new PR for that to keep things clean. But to keep the conversation intact, some question here.

Quoting @MrPetovan 
> Alternatively:
> 
> ```
> $ bin/composer.phar require bower-asset/fork-awesome
> ```
Just for future reference, it's `forkawesome/fork-awesome`.

> 
> The files will be available in `view/asset/fork-awesome`.

Nope, nothing there. Composer is working through quite a few packages and in the end tells me:
```bash
Package operations: 1 install, 0 updates, 0 removals           
  - Installing forkawesome/fork-awesome (1.1.7): Downloading (100%
```

But in ./vendor/ there is forkawesome. Is this the same and can I use it for Frio?

**Comment by MrPetovan** [comment]
No. We use Composer to pull two different types of dependencies:
- PHP dependencies that aren't supposed to be publicly exposed. They are stored in the classic folder `vendor/`.
- Javascript/CSS dependencies that are supposed to be publicly exposed. They are stored in the configured `view/asset/` folder. They use the vendor prefixes `bower-asset` or `npm-asset` because we pull from the Bower or NPM repositories respectively.

Let's say that one day we want to increase Friendica's security and move the main `index.php` to the `view/` folder. At this point the `vendor/` folder won't be publicly accessible from the domain name and any frontend dependency will have to be moved to `view/` at this point.

Instead I chose to prepare in advance for this eventuality by separating the target folders of both dependency types.

So please use the `bower-asset/fork-awesome` package name and link to the files in `view/asset/fork-awesome`.

**Comment by vinzv** [comment]
Alright, now I got it. Thanks for the explanation!

I close this pull request as it's superseded by #7551.

**Pull request closed without merging**